pallets/delegation/src/migrations.rs

```diff
@@ -0,0 +1,480 @@
+				if let Some(root_children_ids) = Children::<T>::take(old_root_id) {
+					new_root_node.children = root_children_ids.iter().copied().collect();
+				}
+				// Add Chilred::take() weight
+				total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+				DelegationHierarchies::insert(old_root_id, new_hierarchy_info);
+				// Adds a read from Roots::drain() and a write from
+				// DelegationHierarchies::insert() weights
+				total_weight = total_weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+				new_nodes.insert(old_root_id, new_root_node);
+			}
+
+			// Then iterate over the regular delegation nodes.
+			for (old_node_id, old_node) in Delegations::<T>::drain() {
+				let new_node_details = DelegationDetails::<T> {
+					owner: old_node.owner,
+					permissions: old_node.permissions,
+					revoked: old_node.revoked,
+				};
+				// In the old version, a parent None indicated the node is a child of the root.
+				let new_node_parent_id = old_node.parent.unwrap_or(old_node.root_id);
+				let mut new_node =
+					DelegationNode::<T>::new_node(old_node.root_id, new_node_parent_id, new_node_details);
+				if let Some(children_ids) = Children::<T>::take(old_node_id) {
+					new_node.children = children_ids.iter().copied().collect();
+				}
+				// Add Chilred::take() weight
+				total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+				// Adds a read from Roots::drain() weight
+				total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+				new_nodes.insert(old_node_id, new_node);
+			}
+
+			// By now, all the children should have been correctly added to the nodes.
+			// We now need to modify all the nodes that are children by adding a reference
+			// to their parents.
+			for (new_node_id, new_node) in new_nodes.clone().into_iter() {
+				// FIXME: new_node.children.iter().cloned() might be possibly changed to
+				// iter_mut.
+				for child_id in new_node.children.iter().cloned() {
+					new_nodes
+						.entry(child_id)
+						.and_modify(|node| node.parent = Some(new_node_id));
+				}
+				// We can then finally insert the new delegation node in the storage.
+				DelegationNodes::<T>::insert(new_node_id, new_node);
+				// Adds a write from DelegationNodes::insert() weight
+				total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+			}
+
+			LastUpgradeVersion::<T>::set(1);
+			// Adds a write from LastUpgradeVersion::set() weight
+			total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+			log::debug!("Total weight consumed: {}", total_weight);
+			log::info!("v0 -> v1 delegation storage migrator finished!");
+			total_weight
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_migrate(&self) -> Result<(), DelegationMigrationError> {
+			ensure!(
+				LastUpgradeVersion::<T>::get() == 1,
+				DelegationMigrationError::MigrationResultInconsistent
+			);
```

Comment:
I would definitely iterate over all delegation nodes and check whether all `(parent, child)` pairs match! I know you are doing unit tests (which I like!) but still.

Reply:
Addressed in https://github.com/KILTprotocol/mashnet-node/commit/25a01bd68312b445ced61688cebebada5861613f.